clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1019,7 +1019,8 @@ public void subscribe(Pattern pattern) {
     public void unsubscribe() {
         acquireAndEnsureOpen();
         try {
-            log.info("Unsubscribed all topics or patterns and assigned partitions");
+            log.info("Unsubscribing all topics or patterns and assigned partitions");
```

Comment:
In general if we want to have a statement that says the event is completed. The original logging `Unsubscribed all topics or patterns and assigned partitions` meets this requirement. It is a bit weird to log the event that is happening without a followup log that says the event is completed.

I agree that the original statement is not accurate by printing `Unsubscribed...` before doing it. Given that `unsubscribe()` is a simple and fast method, there is probably no need for additional INFO level logging to specify whether it is happening. How about we just move the existing INFO log to the end of the try{...} code block without changing it?

Reply:
sounds good.